src/Makefile

```diff
@@ -102,8 +102,11 @@ ifeq (${FC},nvfortran)
             OPTS += -acc -gpu=cc70,lineinfo,nofma,autocompare,math_uniform
             OPTS_M += -acc -gpu=cc70,lineinfo,nofma,autocompare,math_uniform
         else
-            OPTS += -acc -gpu=cc70,lineinfo,nofma,math_uniform -Mpcast -Minfo=accel
-            OPTS_M += -acc -gpu=cc70,lineinfo,nofma,math_uniform,managed -Mpcast -Minfo=accel
+	    # Potentially interesting non-default compiler flags:: 
+	    #     -Minfo=accel    Provides verbose output for accelerator compilation phase
+	    #     -Mpcast         Enables the use of PCAST functionality
+            OPTS += -acc -gpu=cc70,cc80,lineinfo,nofma,math_uniform
+            OPTS_M += -acc -gpu=cc70,cc80,lineinfo,nofma,math_uniform,managed
```

Comment:
Based on my understanding, doing these will make the executable file much larger in size. I have introduced a new compiler option `GPU_TYPE` in the Makefile and it looks like:

```
GPU_TYPE_L    = $(shell echo $(GPU_TYPE) | tr A-Z a-z)

...

ifeq (${USE_OPENACC_L},true)
    ifeq (${DEBUG_L},true)
        ifeq (${GPU_TYPE_L},a100)
            OPTS += -acc -gpu=cc80,lineinfo,nofma,autocompare,math_uniform
            OPTS_M += -acc -gpu=cc80,lineinfo,nofma,autocompare,math_uniform
        else
            OPTS += -acc -gpu=cc70,lineinfo,nofma,autocompare,math_uniform
            OPTS_M += -acc -gpu=cc70,lineinfo,nofma,autocompare,math_uniform
        endif
    else
        ifeq (${GPU_TYPE_L},a100)
            OPTS += -acc -gpu=cc80,lineinfo,nofma,math_uniform -Mpcast -Minfo=accel
            OPTS_M += -acc -gpu=cc80,lineinfo,nofma,math_uniform,managed -Mpcast -Minfo=accel
        else
            OPTS += -acc -gpu=cc70,lineinfo,nofma,math_uniform -Mpcast -Minfo=accel
            OPTS_M += -acc -gpu=cc70,lineinfo,nofma,math_uniform,managed -Mpcast -Minfo=accel
        endif
    endif
    DM       += -D_OPENACC
endif
```

Will this implementation be better?

Reply:
@sjsprecious if you want to go ahead and modify the Makefile either for this PR or for a future one I would be fine with these changes.

Reply:
@johnmauff Thanks John. I have just pushed the changes of Makefile to your branch.